src/compas_timber/ghpython/components/CT_Model/code.py

```diff
@@ -112,71 +115,73 @@ def get_joints_from_rules(self, beams, rules, topologies):
                         )
         return joints
 
-    def RunScript(self, Beams, JointRules, Features, MaxDistance, CreateGeometry):
-        if not Beams:
+    def RunScript(self, beams, joint_rules, features, max_distance, create_geometry):
+        if not beams:
             self.AddRuntimeMessage(Warning, "Input parameter Beams failed to collect data")
-        if not JointRules:
+        if not joint_rules:
             self.AddRuntimeMessage(Warning, "Input parameter JointRules failed to collect data")
-        if not (Beams):  # shows beams even if no joints are found
+        if not (beams):  # shows beams even if no joints are found
             return
-        if MaxDistance is None:
-            MaxDistance = TOL.ABSOLUTE  # compared to calculted distance, so shouldn't be just 0.0
+        if max_distance is None:
+            max_distance = TOL.ABSOLUTE  # compared to calculted distance, so shouldn't be just 0.0
 
-        Model = TimberModel()
         debug_info = DebugInfomation()
-        for beam in Beams:
-            # prepare beams for downstream processing
-            beam.remove_features()
-            beam.remove_blank_extension()
-            beam.debug_info = []
-            Model.add_beam(beam)
-        topologies = []
-        solver = ConnectionSolver()
-        found_pairs = solver.find_intersecting_pairs(Beams, rtree=True, max_distance=MaxDistance)
-        for pair in found_pairs:
-            beam_a, beam_b = pair
-            detected_topo, beam_a, beam_b = solver.find_topology(beam_a, beam_b, max_distance=MaxDistance)
-            if not detected_topo == JointTopology.TOPO_UNKNOWN:
-                topologies.append({"detected_topo": detected_topo, "beam_a": beam_a, "beam_b": beam_b})
-        Model.set_topologies(topologies)
-
-        beams = Model.beams
-        joints = self.get_joints_from_rules(beams, JointRules, topologies)
-
-        if joints:
-            handled_beams = []
-            joints = [j for j in joints if j is not None]
-            # apply reversed. later joints in orginal list override ealier ones
-            for joint in joints[::-1]:
-                beams_to_pair = joint.beams
-                beam_pair_ids = set([id(beam) for beam in beams_to_pair])
-                if beam_pair_ids in handled_beams:
-                    continue
-                try:
-                    joint.joint_type.create(Model, *beams_to_pair, **joint.kwargs)
-                except BeamJoinningError as bje:
-                    debug_info.add_joint_error(bje)
-                else:
-                    handled_beams.append(beam_pair_ids)
+        if create_geometry:
```

Comment:
maybe reverse this logic (i.e. `if not create_geometry: return`)? the indentation is already quite deep plus it makes it hard to see the change in this PR..

Reply:
nah this should only run if `create_geometry` is true, but there is more that needs to run after this.